wrappers/src/results_cs.cpp

```diff
@@ -124,20 +123,16 @@ REALM_EXPORT Results* results_get_filtered_results(const Results& results, uint1
 {
     return handle_errors(ex, [&]() {
         Utf16StringAccessor query_string(query_buf, query_len);
-        auto query = results.get_query();
         auto const &realm = results.get_realm();
 
-        parser::ParserResult result = parser::parse(query_string.to_string());
-
-        parser::KeyPathMapping mapping;
+        query_parser::KeyPathMapping mapping;
         realm::populate_keypath_mapping(mapping, *realm);
 
-        query_builder::NoArguments no_args;
-        query_builder::apply_predicate(query, result.predicate, no_args, mapping);
-
-        DescriptorOrdering ordering;
-        query_builder::apply_ordering(ordering, query.get_table(), result.ordering);
-        return new Results(realm, std::move(query), std::move(ordering));
+        auto table = results.get_table();
```

Comment:
[nit] Since `table` is used only once, we can remove the extra variable here and use `results.get_table().query(...)` on line 133.